components/net/http_cache.rs

```diff
@@ -721,6 +727,11 @@ impl HttpCache {
         assert_eq!(response.status.map(|s| s.0), Some(StatusCode::NOT_MODIFIED));
         let entry_key = CacheKey::new(&request);
         if let Some(cached_resources) = self.entries.get_mut(&entry_key) {
+            trace!(
+                "there are {} cached responses for {:?}",
+                cached_resources.len(),
+                request.url()
+            );
             for cached_resource in cached_resources.iter_mut() {
```

Comment:
A simple, and perhaps effective, fix, for now, might be to sort those by their original expiry. Then we can leave a TODO regarding the full "selection" logic as described at https://tools.ietf.org/html/rfc7234#section-4.3.4

Reply:
On second thought, I think it might be best to be stricter, and use the `Last-Modified` header of the 304 response and find the cached resource whose header has the same value as the 304, and use the body of that resource to construct a new 200 cached response, as described at https://tools.ietf.org/html/rfc7232#section-2.2.2

(As the spec reads, It would be best to only do that if the stored resource has a `Date` header, and the `Last-Modified` header of the 304 response is at least 60 seconds ealier than the store date). 

And the place were we could sort stored resources by their expiry, and take the one that is "valid for the longest time", would be when constructing a response, I think.